storage/entities/base_entity.js

```diff
@@ -123,6 +124,10 @@ class BaseEntity {
 		throw new ImplementationPendingError();
 	}
 
+	getFields() {
+		return Object.keys(this.fields);
+	}
+
```

Comment:
For now `this.fields` is totally for internal use, don't think exposing this information is helpful, rather may confuse people.